authentik/core/models.py

```diff
@@ -616,6 +631,102 @@ def ui_user_settings(self) -> UserSettingSerializer | None:
+        if object_type == User:
+            qs = self.user_property_mappings.all().select_subclasses()
+        elif object_type == Group:
+            qs = self.group_property_mappings.all().select_subclasses()
+        return PropertyMappingManager(
+            qs,
+            self.property_mapping_type,
+            ["source", "properties"] + context_keys,
+        )
+
+    def get_base_user_properties(self, **kwargs) -> dict[str, Any | dict[str, Any]]:
+        """Get base properties for a user to build final properties upon."""
+        raise NotImplementedError
+
+    def get_base_group_properties(self, **kwargs) -> dict[str, Any | dict[str, Any]]:
+        """Get base properties for a group to build final properties upon."""
+        raise NotImplementedError
+
+    def get_base_properties(
+        self, object_type: type[User | Group], **kwargs
+    ) -> dict[str, Any | dict[str, Any]]:
+        """Get base properties for a user or a group to build final properties upon."""
+        if object_type == User:
+            properties = self.get_base_user_properties(**kwargs)
+            properties.setdefault("path", self.get_user_path())
+            return properties
+        if object_type == Group:
+            return self.get_base_group_properties(**kwargs)
+        return {}
+
+    def build_object_properties(
+        self,
+        object_type: type[User | Group],
+        mapper: "PropertyMappingManager | None" = None,
+        user: User | None = None,
+        request: HttpRequest | None = None,
+        **kwargs,
+    ) -> dict[str, Any | dict[str, Any]]:
+        """Build a user or group properties from the source configured property mappings."""
+        from authentik.events.models import Event, EventAction
+
+        properties = self.get_base_properties(object_type, **kwargs)
+        if "attributes" not in properties:
+            properties["attributes"] = {}
+
+        if not mapper:
+            mapper = self.get_mapper(object_type, list(kwargs.keys()))
+        evaluations = mapper.iter_eval(
+            user=user,
+            request=request,
+            return_mapping=True,
+            source=self,
+            properties=properties,
+            **kwargs,
+        )
+        while True:
+            try:
+                value, mapping = next(evaluations)
+            except StopIteration:
+                break
+            except PropertyMappingExpressionException as exc:
+                Event.new(
+                    EventAction.CONFIGURATION_ERROR,
+                    message=f"Failed to evaluate property mapping: '{exc.mapping.name}'",
+                    source=self,
+                    mapping=exc.mapping,
+                ).save()
+                LOGGER.warning(
+                    "Mapping failed to evaluate",
+                    exc=exc,
+                    source=self,
+                    mapping=exc.mapping,
+                )
+                continue
```

Comment:
Mapping exception should stop and not continue

Reply:
done, we now fail with `PropertyMappingExpressionException` if the expression fails to evaluate

The LDAP sync code then catches this and throws StopSync, to preserve existing behaviour.